nimble/host/src/ble_l2cap_coc.c

```diff
@@ -186,9 +186,13 @@ ble_l2cap_coc_rx_fn(struct ble_l2cap_chan *chan)
     rx = &chan->coc_rx;
 
     om_total = OS_MBUF_PKTLEN(*om);
-    rc = ble_hs_mbuf_pullup_base(om, BLE_L2CAP_SDU_SIZE);
-    if (rc != 0) {
-        return rc;
+    /* Continuation LE frames might be of size 1, so we skip the pullup step
+     * for these */
+    if (om_total >= BLE_L2CAP_SDU_SIZE) {
+        rc = ble_hs_mbuf_pullup_base(om, BLE_L2CAP_SDU_SIZE);
```

Comment:
As you noticed that should be done only for the very first packet.
Therefore instead of that `if` please move that code  below the 

``` 
   /* First LE frame */
    if (OS_MBUF_PKTLEN(rx->sdu) == 0) {
```

Btw this pullup might look paranoid but we want to do it before accessing om_data directly.

Reply:
Makes good sense to me, will asap.